bin/boot_docker

```diff
@@ -51,7 +51,8 @@ GOOGLE_PRIVATE_KEY=$GOOGLE_PRIVATE_KEY
 GOOGLE_CLIENT_EMAIL=$GOOGLE_CLIENT_EMAIL
 GOOGLE_CLIENT_ID=$GOOGLE_CLIENT_ID
 GOOGLE_CLOUD_PROJECT=$GOOGLE_CLOUD_PROJECT
-DOCKER_IMAGE_VERSION='latest'
+DOCKER_IMAGE_NAME='gcr.io/broad-singlecellportal-staging/single-cell-portal'
+DOCKER_IMAGE_VERSION='development'
```

Comment:
Blocker: please add code comments that explain why we reference "staging" and "development" in this artifact that (IIUC) is used in production deployment.  I likely misunderstand something, but at a glance and without much context it's alarming.  Code comments here would help allay such concerns.

I don't think I'd appreciated that we use staging namespaces for artifacts that affect production deployment -- e.g. `gcr.io/broad-singlecellportal-staging/rails-baseimage:2.1.0` in the Dockerfile.  That alarms me, because I can imagine someone else on the team presuming "staging" or "development" artifacts don't require a ton of care before deleting, and thus inadvertently breaking our production deployment process.

Ideally, I think, we'd typically omit use the the strings "staging" or "development" in Docker image names and versions referenced in our code.  However, I haven't wrapped my head fully around when these images propagate to Google Cloud Registry / GCR, and I don't have a sense of how feasible it is to omit those strings.  So I think it's worth considering, but I don't insist it's blocking.

In other words, refining the name and version strings would be great, but my only blocking concern is that we need code commentary briefly explaining why we use these "staging" and "development" strings, wherever they're used.

Reply:
Great questions - the TL;DR is that the naming scheme for Google Container Registry is `[service]/[repository]/[image]:[tag]`, where `gcr.io` is the service, `broad-singlecellportal-staging` is the repository, `single-cell-portal` is the image name, and `development` is the tag of the image.  Therefore, we can't can't change the repository, as that's how Docker knows what GCP project to look in.  The argument for using `development` is a reference to the fact that this image will be continually rebuilt to reflect the state of gems/packages required on the `development` branch of `single_cell_portal_core`.  For production releases, the tag would correspond to the version number of the upcoming release, e.g. `1.20.0`.

"Why did we put everything in staging", you ask?  Basically, the main reason was that you cannot specify permission on a per-image basis.  Everything is either public, or private.  At the time, we decided we didn't want a lot of publicly accessible resources hosted out of our production GCP project.  Probably not a real security concern, but that was the decision we made.  After that, it was simply "well, that's where all the other Docker images are..."

I agree that it isn't ideal, but migrating all of our images over to our production project would entail changing a great deal of automation - basically our entire CI/CD pipeline for our 3 main repositories.  That means changes to Jenkins as well as CircleCI/Github Actions jobs, on top of renaming everything in each repository. All that being said, I think it's a completely valid concern - one that I share - and that we should in fact change it, but not in the scope of this PR.  We have bandwidth reserved for tech debt this quarter, and this is a nice project to lump in with that.

Reply:
Very helpful context, and I agree fully resolving this tech debt is out of scope for this PR.  

My only blocking concern here (and elsewhere I note "Blocker" in this PR) is that we need code comments.  Perhaps:

```suggestion
# The "broad-singlecellportal-staging" GCR repository is used in production.  
# The "development" tag is used in non-production deployment.  For production deployment, tag is version number of upcoming release, e.g. 1.20.0.  
# More context: https://github.com/broadinstitute/single_cell_portal_core/pull/1552#discussion_r910424433
DOCKER_IMAGE_NAME='gcr.io/broad-singlecellportal-staging/single-cell-portal'
DOCKER_IMAGE_VERSION='development'
```

Reply:
[9db0be5](https://github.com/broadinstitute/single_cell_portal_core/pull/1552/commits/9db0be51a5d54f69d19138edbe4fe6c7388316fa)